### PR TITLE
Checkout structure added; For validation let’s use https://monterail.…

### PR DESCRIPTION
**Pull request by pkarw**
…github.io/vuelidate/#sub-contextified-validators from Monterail!

### CODE REVIEW

File: src/components/core/blocks/Checkout/OrderReview.vue [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div id='order-review'>
```
Comment (filrak) [inline src/components/core/blocks/Checkout/OrderReview.vue]: should be class

File: src/components/core/blocks/Checkout/OrderReview.vue [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Core order review
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'OrderReview',
+  beforeMount () {
```
Comment (filrak) [inline src/components/core/blocks/Checkout/OrderReview.vue]: beforeMount ()  & components is empty therefore isn't required

File: src/components/core/blocks/Checkout/Payment.vue [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<template>
```
Comment (filrak) [inline src/components/core/blocks/Checkout/Payment.vue]: same as OrderReview.vue

File: src/components/core/blocks/Checkout/PersonalDetails.vue [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<template>
```
Comment (filrak) [inline src/components/core/blocks/Checkout/PersonalDetails.vue]: same as OrderReview.vue

File: src/components/core/blocks/Checkout/Shipping.vue [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<template>
```
Comment (filrak) [inline src/components/core/blocks/Checkout/Shipping.vue]: same as OrderReview.vue

File: src/pages/Checkout.vue [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div id='checkout'>
+    Core Checkout
+  </div>
+</template>
+
+<script>
+import PersonalDetails from '../components/core/blocks/Checkout/PersonalDetails.vue'
```
Comment (filrak) [inline src/pages/Checkout.vue]: In new components we should use aliases from webpack

File: src/themes/default/components/core/blocks/Checkout/OrderReview.vue [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    Core order review
+  </div>
+</template>
+
+<script>
+import { coreComponent } from 'lib/themes'
+
+export default {
+  mixins: [coreComponent('core/blocks/Checkout/OrderReview')],
+  beforeMount () {
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Checkout/OrderReview.vue]: unnecessary code, empty objects

File: src/pages/Checkout.vue [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import Shipping from '../components/core/blocks/Checkout/Shipping.vue'
+import Payment from '../components/core/blocks/Checkout/Payment.vue'
+import OrderReview from '../components/core/blocks/Checkout/OrderReview.vue'
+import Microcart from '../components/core/blocks/Microcart/Microcart.vue'
+
+export default {
+  name: 'Checkout',
+  beforeMount () {
```
Comment (filrak) [inline src/pages/Checkout.vue]: unnecessary empty function

File: src/themes/default/components/core/blocks/Checkout/Payment.vue [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<template>
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Checkout/Payment.vue]: same as OrderReview.vue

File: src/themes/default/components/core/blocks/Checkout/PersonalDetails.vue [hunk 10]
```diff
@@ -0,0 +1,23 @@
+<template>
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Checkout/PersonalDetails.vue]: same as OrderReview.vue

File: src/themes/default/components/core/blocks/Checkout/Shipping.vue [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<template>
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Checkout/Shipping.vue]: same as OrderReview.vue

File: src/themes/default/pages/Checkout.vue [hunk 12]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <div id='checkout'>
+    <personal-details></personal-details>
+    <shipping></shipping>
+    <payment></payment>
+    <order-review></order-review>
+    <microcart isCheckoutMode="true"></microcart>
```
Comment (filrak) [inline src/themes/default/pages/Checkout.vue]: What isCheckoutMode is for? if you want conditional layout changes for Microcart component allowing to put it in checkout page it's a bad idea. 9 think we should just use different component for this.

Comment (pkarw) [reply src/themes/default/pages/Checkout.vue]: Ok, so i believe we should create /Checkot/Microcart which will have Microcart/Microcart among the mixins. WDYT?

File: src/components/core/blocks/Checkout/OrderReview.vue (hunk 1 above)
Comment (pkarw) [reply src/components/core/blocks/Checkout/OrderReview.vue]: OK

File: src/components/core/blocks/Checkout/OrderReview.vue (hunk 2 above)
Comment (pkarw) [reply src/components/core/blocks/Checkout/OrderReview.vue]: OK

File: src/components/core/blocks/Checkout/Payment.vue (hunk 3 above)
Comment (pkarw) [reply src/components/core/blocks/Checkout/Payment.vue]: OK

File: src/components/core/blocks/Checkout/PersonalDetails.vue (hunk 4 above)
Comment (pkarw) [reply src/components/core/blocks/Checkout/PersonalDetails.vue]: OK

File: src/components/core/blocks/Checkout/Shipping.vue (hunk 5 above)
Comment (pkarw) [reply src/components/core/blocks/Checkout/Shipping.vue]: OK

File: src/pages/Checkout.vue (hunk 6 above)
Comment (pkarw) [reply src/pages/Checkout.vue]: OK

File: src/pages/Checkout.vue (hunk 8 above)
Comment (pkarw) [reply src/pages/Checkout.vue]: OK

File: src/themes/default/components/core/blocks/Checkout/OrderReview.vue (hunk 7 above)
Comment (pkarw) [reply src/themes/default/components/core/blocks/Checkout/OrderReview.vue]: OK

File: src/themes/default/components/core/blocks/Checkout/Payment.vue (hunk 9 above)
Comment (pkarw) [reply src/themes/default/components/core/blocks/Checkout/Payment.vue]: OK

File: src/themes/default/components/core/blocks/Checkout/PersonalDetails.vue (hunk 10 above)
Comment (pkarw) [reply src/themes/default/components/core/blocks/Checkout/PersonalDetails.vue]: OK

File: src/themes/default/components/core/blocks/Checkout/Shipping.vue (hunk 11 above)
Comment (pkarw) [reply src/themes/default/components/core/blocks/Checkout/Shipping.vue]: OK

**Comment by pkarw** [comment]
@filrak fixes made in https://github.com/DivanteLtd/vue-storefront/pull/192/commits/aca726dd6927a66e1daefe8511a9f21f4e097670

**Pull request merged**